File: lentaru-server/src/com/xeppaka/lentaruserver/items/body/LentaBodyItemImage.scala
package com.xeppaka.lentaruserver.items.body

import com.xeppaka.lentaruserver.items.ItemBase
import java.net.{URLDecoder, URLEncoder}
import java.net
import org.apache.commons.lang3.StringEscapeUtils

/**
 * Created with IntelliJ IDEA.
 * User: nnm
 * Date: 11/16/13
 * Time: 5:49 PM
 * To change this template use File | Settings | File Templates.
 */
case class LentaBodyItemImage(val preview_url: String, val original_url: String, val caption: Option[String], val credits: Option[String]) extends ItemBase {
  def toXml(indent: String): String = {
    val sb = new StringBuilder(indent)
    sb.append("<image preview_url=\"").append(preview_url).append("\" ")
    sb.append("original_url=\"").append(original_url).append("\" ")

    caption match {
      case Some(value) => sb.append("caption=\"").append(StringEscapeUtils.escapeXml(StringEscapeUtils.unescapeXml(value))).append("\" ")
      case None =>
    }

    credits match {
      case Some(value) => sb.append("credits=\"").append(StringEscapeUtils.escapeXml(StringEscapeUtils.unescapeXml(value))).append("\" ")
      case None =>
    }

    sb.append("/>\n").toString()
  }

  override def hashCode(): Int = {
    var hash = 1
    hash = hash * 31 + preview_url.hashCode
    hash = hash * 31 + original_url.hashCode
    hash = hash * 31 + caption.hashCode
    hash = hash * 31 + credits.hashCode

    hash
  }

  override def equals(obj: scala.Any): Boolean = {
    if (!obj.isInstanceOf[LentaBodyItemImage])
      false
    else {
      val other = obj.asInstanceOf[LentaBodyItemImage]
      preview_url == other.preview_url && original_url == other.original_url && caption == other.caption && credits == other.credits
    }
  }
}

File: lentaru/lentareader/src/main/java/com/xeppaka/lentareader/data/body/items/LentaBodyItemImage.java
package com.xeppaka.lentareader.data.body.items;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.text.TextUtils;

import com.xeppaka.lentareader.ui.widgets.fullnews.FullNewsElement;

import org.apache.commons.lang3.StringEscapeUtils;

import java.io.Serializable;

/**
 * Created by nnm on 11/18/13.
 */
public class LentaBodyItemImage implements Item, Serializable {
    private String preview_url;
    private String original_url;
    private String caption;
    private String credits;

    public LentaBodyItemImage(String preview_url, String original_url, String caption, String credits) {
        this.preview_url = preview_url;
        this.original_url = original_url;
        this.caption = caption;
        this.credits = credits;
    }

    public String getPreview_url() {
        return preview_url;
    }

    public String getOriginal_url() {
        return original_url;
    }

    public String getCaption() {
        return caption;
    }

    public String getCredits() {
        return credits;
    }

    @Override
    public String toXml() {
        StringBuilder sb = new StringBuilder("<image preview_url=\"");
        sb.append(preview_url).append("\" ").append("original_url=\"").append(original_url).append("\" ");

        if (caption != null) {
            sb.append("caption=\"").append(StringEscapeUtils.escapeXml(caption)).append("\" ");
        }

        if (credits != null) {
            sb.append("credits=\"").append(StringEscapeUtils.escapeXml(credits)).append("\" ");
        }

        return sb.append("/>").toString();
    }

    @Override
    public FullNewsElement createFullNewsListElement(Context context, Fragment fragment) {
        return null;
    }

    public boolean hasCaption() {
        return caption != null && !TextUtils.isEmpty(caption);
    }

    public boolean hasCredits() {
        return credits != null && !TextUtils.isEmpty(credits);
    }
}
